Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Helpers/SmartWeakEvent.cs

```diff
@@ -0,0 +1,174 @@
+﻿// ******************************************************************
```

Comment:
Do you think you can leverage the WeakEventListener: https://github.com/Microsoft/UWPCommunityToolkit/blob/dev/Microsoft.Toolkit.Uwp/Helpers/WeakEventListener.cs ?